### PR TITLE
chore: skip process auth admin check if rbac is enable

### PR DESCRIPTION
**Pull request by corban-beaird**
## Description
If RBAC is enabled, we don't want to check if the admin flag in `ProcessAuthentication`, this PR enables provides functionality to check if RBAC is enabled based on the authz provider type. 

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/internal/config/authz_config.go [hunk 1]
```diff
@@ -86,6 +90,26 @@ func (c AuthZConfig) IsRBACUIEnabled() bool {
 	return c.Type != BasicAuthZType
 }
 
+// IsRBACEnabled returns if the authz config type is using the RBAC implementation
+// and will attempt to use Fallback and Default types if necessary.
+func (c AuthZConfig) IsRBACEnabled() bool {
```
Comment (hamidzr) [inline master/internal/config/authz_config.go:95]: suggestion: I'd put it next to isrbacuienabled and differentiate them

**Review by hamidzr — APPROVED** [review]
🚢

File: master/internal/config/authz_config.go (hunk 1 above)
Comment (corban-beaird) [reply master/internal/config/authz_config.go:95]: @hamidzr what do you mean by next to it, like an additional comment? This is right below `IsRBACUIEnabled`

Comment (hamidzr) [reply master/internal/config/authz_config.go:95]: my bad I only checked lines coming after this so I didn't realize it's right below it. all good

**Pull request merged**